File: ObjectAvoidance/java/src/main/java/elite/math/trig/Cosecant.java
package elite.math.trig;

import elite.math.trig.triangle.Hypotenuse;
import elite.math.trig.triangle.Opposite;

/**
 *
 * @author CFerg (Elite)
 */
public class Cosecant {

    /**
     *
     */
    public double cosecant;
    
    /**
     *
     * @param h
     * @param o
     */
    public Cosecant(Hypotenuse h, Opposite o){
        this.cosecant = (h.hypotenuse / o.opposite);
    }
    
    /**
     *
     * @param arccos
     */
    public Cosecant(Arccosecant arccos){
        this.cosecant = (1 / arccos.arccosecant);
    }
    
    /**
     *
     * @param sin
     */
    public Cosecant(Sine sin){
        this.cosecant = (1 / sin.sine);
    }
}

File: ObjectAvoidance/src/elite/math/trig/Cosecant.kt
package elite.math.trig

import elite.math.trig.triangle.Hypotenuse
import elite.math.trig.triangle.Opposite

/**
 *
 * @author CFerg (Elite)
 */
class Cosecant {

    /**
     *
     */
    var cosecant: Double = 0.toDouble()

    /**
     *
     * @param h
     * @param o
     */
    constructor(h: Hypotenuse, o: Opposite) {
        this.cosecant = h.hypotenuse / o.opposite
    }

    /**
     *
     * @param arccos
     */
    constructor(arccos: Arccosecant) {
        this.cosecant = 1 / arccos.arccosecant
    }

    /**
     *
     * @param sin
     */
    constructor(sin: Sine) {
        this.cosecant = 1 / sin.sine
    }
}